subsys/bluetooth/host/settings.c

```diff
@@ -100,8 +117,41 @@ void bt_settings_encode_key(char *path, size_t path_size, const char *subsys,
 	}
 
 	LOG_DBG("Encoded path %s", path);
+
+	return 0;
 }
-#endif
+
+static int bt_settings_encode_key_no_addr(char *path, size_t path_size, const char *key)
+{
+	size_t len = 3;
+
+	/* Skip if path_size is less than 3; strlen("bt/") */
+	if (len < path_size) {
+		/* Key format:
+		 *  "bt/<key>", "/<key>" is optional
+		 */
+		strcpy(path, "bt/");
+
+		/* Concatenate key as much the free space permits */
+		strncpy(&path[len], key, path_size - len);
+		len = strlen(path);
+
+		/* If path string is full, always null terminate at path_size */
+		if (len >= path_size) {
+			/* Truncate string */
+			path[path_size - 1] = '\0';
+		}
+
+	} else if (path_size > 0) {
+		/* path_size not sufficient for "bt/" */
+		*path = '\0';
+	}
```

Comment:
Should this be an assert? Seems like a programming issue, rather than a settings-content issue